ocaml/xapi/xapi_pool.ml

```diff
@@ -3021,6 +3028,10 @@ let get_updates_handler (req : Http.Request.t) s _ =
                when (List.mem failure failures_of_404) ->
                  error "404: can't get updates for pool: %s" (ExnHelper.string_of_exn e);
                  Http_svr.headers s (Http.http_404_missing ())
+             | Api_errors.(Server_error (failure, _)) as e
+               when not (List.mem failure failures_of_404) ->
+                 error "getting updates for pool failed: %s" (ExnHelper.string_of_exn e);
+                 raise e
```

Comment:
Is this the right exception to raise? Below a different one is raised for a similar case.

Reply:
I think that this one here is already a proper API error, which can be re-raised. Below is a different exception, which needs to be turned into an API error.